src/extension/lsp.ts

```diff
@@ -49,6 +50,27 @@ async function fileExists(pathFromRoot: string): Promise<boolean> {
   }
 }
 
+async function applyAllFixes() {
+  const textEditor = window.activeTextEditor
+  if (!textEditor) {
+    return
+  }
+  const textDocument = {
+    uri: textEditor.document.uri.toString(),
+    version: textEditor.document.version,
+    text: textEditor.document.getText(),
+  }
+  const params = {
+    command: 'ast-grep.applyAllFixes',
+    arguments: [textDocument],
+  }
+  client.sendRequest(ExecuteCommandRequest.type, params).then(undefined, () => {
+    void window.showErrorMessage(
+      'Failed to apply Ast-grep fixes to the document. Please consider upgrading ast-grep version or opening an issue with steps to reproduce.',
```

Comment:
is void needed here? 
Also, showErrorMessage can have an action button to guide users installing the latest ast-grep version. (say by guiding users to the installing doc page.)

Reply:
done